core/src/main/java/com/yahoo/oak/InternalOakBasics.java

```diff
@@ -355,6 +338,8 @@ public boolean hasNext() {
             return (state != null);
         }
 
+        protected abstract void initStateWithNotDeletedChunk(BasicChunk chunk);
```

Comment:
Please add a comment explaining what is this (new) function, why is it needed, when is it used, what should it generally do?